TMS.Net07.Homework5.FibonacciAndClasses/TMS.Net07.Homework5.Task1. FibonacciArray/TMS.Net07.Homework5.Task1. FibonacciArray/Program.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+
+namespace TMS.Net07.Homework5.Task1.FibonacciArray
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            while (true)
+            {
+                Console.Write("Input number for Fibonacci calculation: ");
+                long inputNumber = 0;
+                bool isTryInputNumber = false;
+
+                do
+                {
+                    isTryInputNumber = long.TryParse(Console.ReadLine(), out inputNumber);
+
+                    if (isTryInputNumber == false)
+                    {
+                        Console.Write("Incorrect input. Repeat, please: ");
+                    }
+                }
+                while (isTryInputNumber == false);
+
+                bool isNumberBelowZero = false;
+
+                if (inputNumber < 0)   // If inputted number <0 , so need to multiply number on -1 to make the calculation more simple.
+                {
+                    inputNumber *= -1;
+                    isNumberBelowZero = true;
+                }
+
+                long[] arrayFibonacciNumbers = new long[inputNumber + 1];
+
+                long result = CalculateFibonacciIteration(inputNumber, arrayFibonacciNumbers);
+
+                if ((isNumberBelowZero == true) && (inputNumber % 2 == 0))
+                {
+                    result *= -1;
```

Comment:
From the user experience, it's usually better to just write a message that the input is wrong than try to fix it. Users can come up with really weird input and it's usually unreal to take care of all cases.